File: 3-4주차/B1260.py
import sys
from collections import deque
input = sys.stdin.readline

N, M, V = map(int, input().split())
graph = [[] for _ in range(N+1)]
for _ in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for adj in graph:
    adj.sort()

visited = [False] * (N+1)

def dfs(start):
    visited[start] = True
    print(start, end=' ')
    for nxt in graph[start]:
        if not visited[nxt]:
            dfs(nxt)

def bfs(start):
    queue = deque([start])
    visited[start] = True
    while queue:
        node = queue.popleft()
        print(node, end=' ')
        for nxt in graph[node]:
            if not visited[nxt]:
                visited[nxt] = True
                queue.append(nxt)

dfs(V)
print()
visited = [False] * (N+1)
bfs(V)

File: 3-4주차/B1260.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, M, V] = input[0].split(' ').map(Number);
const graph = Array.from({ length: N + 1 }, () => []);
for(let i=1; i<=M; i++){
    const [a, b] = input[i].split(' ').map(Number);
    graph[a].push(b);
    graph[b].push(a);
}
for(let i=1; i<=N; i++){
    graph[i].sort((a, b) => a - b);
}

const visited = new Array(N + 1).fill(false);

function dfs(node){
    visited[node] = true;
    process.stdout.write(node + ' ');
    for(const nxt of graph[node]){
        if(!visited[nxt]){
            dfs(nxt);
        }
    }
}

function bfs(start){
    const queue = [];
    queue.push(start);
    visited[start] = true;

    while(queue.length > 0){
        const node = queue.shift();
        process.stdout.write(node + ' ');
        for(const nxt of graph[node]){
            if(!visited[nxt]){
                visited[nxt] = true;
                queue.push(nxt);
            }
        }
    }
}

dfs(V);
console.log();
visited.fill(false);
bfs(V);
